intro/intro_tutorial.py

```diff
@@ -0,0 +1,248 @@
+# Snorkel then automatically models, cleans, and integrates the resulting training data using novel, theoretically-grounded techniques.
+
+# %% [markdown]
+# <img src="img/snorkel_ops.png" onerror="this.onerror=null; this.src='/doks-theme/assets/images/layout/TrainingData.png';" align="center" style="display: block; margin-left: auto; margin-right: auto;">
+
+# %% [markdown]
+# In this quick walkthrough, we'll preview the high level workflow and interfaces of Snorkel using a canonical machine learning problem: classifying spam.
+# We'll use a public [YouTube comments dataset](https://www.kaggle.com/goneee/youtube-spam-classifiedcomments) from Kaggle, and see how **Snorkel can enable you to train a machine learning model without _any_ hand-labeled training data!**
+# For the more detailed version, see the [Introductory Tutorial](#).
+#
+# We'll walk through five basic steps:
+#
+# 1. **Writing Labeling Functions (LFs):** First, rather than hand-labeling any training data, we'll programamtically label our _unlabeled_ dataset with LFs
+# 2. **Modeling & Combining LFs:** Next, we'll use the `LabelModel` to automatically learn the accuracies of our LFs and reweight and combine their outputs
+# 3. **Writing Transformation Functions (TFs) for Data Augmentation:** Then, we'll augment this labeled training set by writing a simple TF
+# 4. **Writing _Slicing Functions (SFs)_ for Data Subset Selection:** Then, we'll write an SF to identify a critical subset or _slice_ of our training set.
+# 5. **Training a final ML model:** Finally, we'll train a simple ML model with our training set!
+#
+# We'll start first by loading the _unlabeled_ comments, which we'll use as our training data, as a pandas `DataFrame`:
+
+# %%
+from utils import load_unlabeled_spam_dataset
+
+df_train = load_unlabeled_spam_dataset()
+
+# %% [markdown]
+# ## (1) Writing Labeling Functions
+#
+# _Labeling functions (LFs)_ are one of the core operators for building and managing training datasets programmatically in Snorkel.
+# The basic idea is simple: **a labeling function is a function that labels some subset of the training dataset**.
+# That is, in our example here, each labeling function takes as input a comment object, and either outputs a label (`SPAM = 1` or `NOT_SPAM = 0`) or abstains from labeling (`ABSTAIN = -1`):
+
+# %%
+# Define the label mappings for convenience
+ABSTAIN = -1
+NOT_SPAM = 0
+SPAM = 1
+
+# %% [markdown]
+# Labeling functions can be used to represent many heuristic and/or noisy strategies for labeling data, often referred to as [weak supervision](#).
+# The basic idea of labeling functions, and other programmatic operators in Snorkel, is to let users inject domain information into machine learning models in higher level, higher bandwidth ways than manually specifying single-bit labels for individual data points.
```

Comment:
"manually specifying single-bit labels for individual data points" is too fancy for "labeling individual examples by hand"